chia/plotting/util.py

```diff
@@ -78,8 +78,12 @@ def get_plot_filenames(root_path: Path) -> Dict[Path, List[Path]]:
     config = load_config(root_path, "config.yaml")
     recursive_scan: bool = config["harvester"].get("recursive_plot_scan", False)
     for directory_name in get_plot_directories(root_path, config):
-        directory = Path(directory_name).resolve()
-        all_files[directory] = get_filenames(directory, recursive_scan)
+        try:
+            directory = Path(directory_name).resolve()
+        except (OSError, RuntimeError):
+            log.exception(f"Failed to resolve {directory_name}")
+        else:
+            all_files[directory] = get_filenames(directory, recursive_scan)
```

Comment:
it seems quite odd to me to use the `else` clause here. partly because it's an odd feature in python, but also because it defeats one of the main purposes of using exceptions. You can just move this line immediately following line 82, and you'll have (almost) the same behavior.

The main difference is that if `get_filenames()` throws, you still continue looping. I think that's better behavior. With this patch, `get_filenames()` throwing will terminate the loop, which I don't think is deliberate.

Reply:
What's the `RuntimeError` that can happen there?  Indicating the error would be great.  Not found, permissions, etc.  Also the directory that's having trouble.  Otherwise I have no information to take action on.

Without saying it shouldn't be done here, I think that handling exceptions from `get_filenames()` is a separate enhancement.  What's here addresses the directly reported failure.  But yeah, handling some more exceptions would probably be nice while we're here.

Putting them in the same `try:` block means you can't differentiate the source of the error and have to provide a generic message that covers both.  I might do separate `try:`/`except:` blocks with a continue in the first `except:` after logging?  Or both for consistency.

Reply:
> What's the `RuntimeError` that can happen there? Indicating the error would be great. Not found, permissions, etc. Also the directory that's having trouble. Otherwise I have no information to take action on.

https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve

See the docs for the `RuntimeError` and since in the linked issue we get the `PermissionError` which is a `OSError` i just picked `OSError` in case similar related errors coming up in `Path.resolve`.


@arvidn @altendky About your other concerns: I moved the `get_filenames` out there, yeah, to just address the actual issue. At least from my understanding and in its current implementation `get_filenames` doesn't raise really. The alternative then to have a `else` is to have a `continue` after the logging to have the correct behaviour. At the end i think the related code here can improved in various places and could be rewritten to not catch some exceptions in the one method and few in the other because it's all kind of related anyway but yeah, just trying to address the linked issue here and not open any rabbit holes :) Do we really need a change here? Let me know.

Reply:
I now see the `except Exception as e:` and...  `*runs and hides under a rock*`  so yeah.  I think reporting the directory that had an issue is important and reporting the error is important.  I think a `continue` fits better here since it avoids the pattern of nesting code in a post-`try:` block.

Maybe wait for Arvid to weigh in so we don't make you flip flop a bunch.

Reply:
logging the error makes sense, and I think the current logging does that well. I still would like some justification for using the `else` block on a `try` block. it adds an extra line of code and I think it also adds confusion.

Reply:
@arvidn I changed it in a9d1621ef1d63286deb51e06cf9382a8866b617b so we can just merge it but it doesn't really add a line of code if we want to distinguish the failure vs having a general error message, i.e. we only catch errors in `resolve` here. Its:

```python
try:
    directory = Path(directory_name).resolve()
except (OSError, RuntimeError):
    log.exception(f"Failed to resolve {directory_name}")
else:
    all_files[directory] = get_filenames(directory, recursive_scan)
```

vs 

```python
try:
    directory = Path(directory_name).resolve()
except (OSError, RuntimeError):
    log.exception(f"Failed to resolve {directory_name}")
    continue
all_files[directory] = get_filenames(directory, recursive_scan)
```

and the `else` is not more confusing than having a `continue` imo, actually i think the first block looks cleaner. I mean.. like other language specific things, if you got it, you got it and "if it doesn't raise, do this instead" doesn't sound that confusing, does it?

Reply:
in my mind, the natural way of writing this would be:
```
try:
    directory = Path(directory_name).resolve()
    all_files[directory] = get_filenames(directory, recursive_scan)
except (OSError, RuntimeError):
    log.exception(f"Failed to resolve {directory_name}")
```

Reply:
The exception handling is intended only for the `.resolve()` call.  We expect `get_filenames()` to handle it's own errors presently (the code seems to).  If we put both things in the `try:` then we have to have a more generic error.  Also, as is, we still don't know if the path doesn't exist, or it's a permission error, or...  something more arcane.

Reply:
I see

Reply:
I was overlooking the use of `log.exception()` that will include the information I'm interested in.  Sorry for that.